modules/ztp-configuring-hwevents-using-pgt.adoc

```diff
@@ -0,0 +1,78 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/ztp-deploying-disconnected.adoc
+
+:_content-type: PROCEDURE
+[id="ztp-creating-hwevents_{context}"]
+= Configuring bare-metal events monitoring using PolicyGenTemplate CRs
+
+You can configure bare-metal hardware events for vRAN clusters that are deployed using the GitOps Zero Touch Provisioning (ZTP) pipeline.
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in as a user with `cluster-admin` privileges.
+* Install the {redfish-operator}.
+* Create a `BMCEventSubscription` CR for the BMC Redfish hardware.
+* Create a Git repository where you manage your custom site configuration data.
+
+[NOTE]
+====
+Multiple `HardwareEvent` resources are not permitted.
+====
+
+.Procedure
+
+. Create the `HardwareEvent` custom resource (CR) entry in the `PolicyGenTemplate`:
+
+.. Save the following YAML in the `hw-event.yaml` file:
++
+[source,yaml]
+----
+apiVersion: event.redhat-cne.org/v1alpha1
+kind: HardwareEvent
+metadata:
+  name: openshift-hw-events
+  namespace: openshift-hw-events
+  annotations:
+    ran.openshift.io/ztp-deploy-wave: "10"
+spec:
+  nodeSelector: {}
+  transportHost: "amqp://amq-router.amq-router.svc.cluster.local"
+  logLevel: "trace"
+----
+
+.. Create the `HardwareEvent` CR by running the following command:
++
+[source,terminal]
+----
+$ oc create -f hw-event.yaml
+----
+
+. Update the `spec.sourceFiles` entry in the `PolicyGenTemplate` with `HadrwareEvent` CR file that configures the AMQ host to the `hardware-events-config`.
```

Comment:
Add the following YAML into .spec.sourceFiles in the common-ranGen.yaml file to configure the Bare Metal Event Rely Operator:
# AMQ interconnect operator for fast events
- fileName: AmqSubscriptionNS.yaml
  policyName: "subscriptions-policy"
- fileName: AmqSubscriptionOperGroup.yaml
  policyName: "subscriptions-policy"
- fileName: AmqSubscription.yaml
  policyName: "subscriptions-policy"
# Bare Metal Event Rely operator
- fileName: HardwareEventSubscriptionNS.yaml
  policyName: "subscriptions-policy"
- fileName: HardwareEventSubscriptionOperGroup.yaml
  policyName: "subscriptions-policy"
- fileName: HardwareEventSubscription.yaml
  policyName: "subscriptions-policy"

Apply the following PolicyGenTemplate changes to your specific site YAML files, for example, example-sno-site.yaml.
In spec.sourceFiles, add the CRs for AMQ Interconnect and HardwareEvent. Make sure the `transportHost` in HardwareEvent match the name and namespace of AMQ Interconnect.
- fileName: AmqInstance.yaml
  policyName: "config-policy"
 - fileName: HardwareEvent.yaml
   policyName: "config-policy"
   spec:
     nodeSelector: {}
     transportHost: "amqp://amq-router.amq-router.svc.cluster.local"
     logLevel: "debug"

Reply:
@jzding I added these steps. I rephrased the second one a bit. Please take a look!